fetchai/ledger/api/tx.py

```diff
@@ -1,6 +1,45 @@
+import base64
+
 from .common import ApiEndpoint
 
 
+class TxStatus:
+    _SUCCESSFUL_TERMINAL_STATES = ('Executed', 'Submitted')
+    _NON_TERMINAL_STATES = ('Unknown', 'Pending')
+
+    def __init__(self,
+                 digest: bytes,
+                 status: str,
+                 exit_code: int,
+                 charge: int,
+                 charge_rate: int,
+                 fee: int):
+        self._digest = digest
+        self._digest_hex = self._digest.hex()
+        self.status = status
+        self.exit_code = exit_code
+        self.charge = charge
+        self.charge_rate = charge_rate
+        self.fee = fee
+
+    @property
+    def successful(self):
+        return self.status in self._SUCCESSFUL_TERMINAL_STATES
+
+    @property
+    def failed(self):
+        return self.status not in self._NON_TERMINAL_STATES and \
+               self.status not in self._SUCCESSFUL_TERMINAL_STATES
+
+    @property
+    def digest(self):
+        return self._digest_hex
+
+    @property
+    def digest_bytes(self):
+        return self._digest
```

Comment:
This is confusing, could we match the internal naming scheme (_digest, _digest_hex) to the external one? Most simply by renaming _digest to _digest_bytes (and possibly digest to digest_hex)?

Reply:
Done.